### PR TITLE
Fix docker base image gitlab job

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
* Change image path: from system-tests to dd-trace-java
* Remove manual trigger in order to avoid not ending checks in pull request.
* Change to a schedule job

### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -48,16 +48,15 @@ deploy_to_reliability_env:
 tracer-base-image:
   extends: .ci_authenticated_job
   rules:
-    - if: '$CI_COMMIT_BRANCH == "master"'
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
```
Comment (pierotibou) [inline .gitlab-ci.yml:51]: Doesn't it make more sense to upload from master rather than on a trigger? Why the change?

Comment (robertomonteromiguel) [reply .gitlab-ci.yml:51]: I am not sure when the trigger is launched.....When the GitLab trigger is launched, CircleCI build is finished??
I am worried that things like this could happen: https://github.com/DataDog/dd-trace-js/issues/2677#issuecomment-1426198898

Comment (pawelchcki) [reply .gitlab-ci.yml:51]: @pierotibou Since we're fetching artifact from circleci last job. We currently don't have any means to wait for circle ci job to finish - to fetch artifact.

At this point having a schedule and uploading last master circleci job periodically makes this a little bit more predictable and will cause somewhat lower traffic overall.

**Review by pawelchcki — APPROVED** [review]

**Review by pierotibou — APPROVED** [review]
